core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2644,19 +2644,22 @@ public void updateMount(AlluxioURI alluxioPath, MountContext context)
       }
       if (syncPoints != null && !syncPoints.isEmpty()) {
         // add back all sync points
-        Exception syncEx = null;
+        InvalidPathException syncEx = null;
         for (AlluxioURI syncPoint : syncPoints) {
           try {
             startSyncAndJournal(rpcContext, syncPoint);
           } catch (Exception e) {
             if (syncEx == null) {
-              syncEx = new UnexpectedAlluxioException("Some sync points in this mount point"
+              syncEx = new InvalidPathException("Some sync points in this mount point"
                   + " failed to initialize. Please add them manually.");
             }
             syncEx.addSuppressed(new InvalidPathException(
                 String.format("Failed to add sync point %s", syncPoint), e));
           }
         }
+        if (syncEx != null) {
+          throw syncEx;
```

Comment:
I'm wary of relying on the exception to hint that there was a partial success, especially if another `InvalidPathException` could mean that the entire operation failed.

Reply:
Do we have a convention on how non-fatal warnings are returned?